golang/cosmos/x/swingset/swingset.go

```diff
@@ -0,0 +1,75 @@
+package swingset
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+	vstoragetypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// portHandler implements a vm.PortHandler.
+type portHandler struct {
+	keeper Keeper
+}
+
+type swingsetMessage struct {
+	Type    string          `json:"type"`
+	Request json.RawMessage `json:"request"`
+}
+
+type swingsetStoreSetRequest struct {
+	Entries []json.RawMessage `json:"entries"`
+}
+
+const (
+	SWINGSET_STORE_SET = "SWINGSET_STORE_SET"
+)
+
+// NewPortHandler returns a port handler for the Keeper.
+func NewPortHandler(k Keeper) vm.PortHandler {
+	return portHandler{keeper: k}
+}
+
+// Receive implements the vm.PortHandler method.
+// Receives and processes a bridge message, returning the
+// JSON-encoded response or error.
+func (ph portHandler) Receive(ctx *vm.ControllerContext, str string) (string, error) {
```

Comment:
In a perfect world I agree unit tests would be good here, but since no other port handler has unit tests, and that this does very little, I felt it was ok to skip.